arch/arm/include/kernel_arch_func.h

```diff
@@ -97,7 +97,9 @@ _arch_switch_to_main_thread(struct k_thread *main_thread,
 		*/
 		"mov %%r0, %3 \t\n"
 		"push {r2, lr} \t\n"
+		"push {r3, lr} \t\n"
```

Comment:
perhaps we should just do what we do in arch/arm/core/userspace.S and do a:
push {r0, r1, r2, r3, r12, lr}
followed by:
pop {r0, r1, r2, r3, r12, lr}

per ARMS calling conventions and the 8 byte stack alignment.